### PR TITLE
fix: Make status check for Wi-Fi on adding device

### LINKED ISSUE #1598: Adding device doesn't checks for wifi enabled
**Actual Behaviour**

When adding device, it simply shows no device present without checking that android wifi is enabled or not. (In hotspot mode also shows device not present)

**Expected Behaviour**

It should check the wifi is enabled or not and then should show the result.

**Steps to reproduce it**

Go to skill listing, then go to settings and then click on add device.

**Screenshots of the issue**

No device found 

**Would you like to work on the issue?**

Yes!

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #1598 

Changes: Added a check method for wifi enabled nd a dialog for it.

### CODE REVIEW

**Comment by codedsun** [comment]
@arundhati24 @batbrain7 Please checkk!

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -364,13 +364,15 @@
...
-    <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby wifi networks available.</string>
+    <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby Wi-Fi networks available.</string>
     <string name="set_up">SET UP</string>
     <string name="device_help">Choose the Wi-Fi network you would like to use with your smart speaker.</string>
     <string name="connecting_device">Connecting to your device...</string>
     <string name="device_name">SUSI.AI</string>
     <string name="connection_success">Connected Successfully !</string>
     <string name="choose_wifi">Connect to Wi-Fi</string>
+    <string name="wifi_access">Provide Wi-Fi Access</string>
+    <string name="wifi_access_message">Please turn on the Wi-Fi in order to search for the nearby Wi-Fi networks available. </string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: Remove space at the end of the string.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -364,13 +364,15 @@
...
     <string name="location_access">Provide Location Access</string>
-    <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby wifi networks available.</string>
+    <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby Wi-Fi networks available.</string>
     <string name="set_up">SET UP</string>
     <string name="device_help">Choose the Wi-Fi network you would like to use with your smart speaker.</string>
     <string name="connecting_device">Connecting to your device...</string>
     <string name="device_name">SUSI.AI</string>
     <string name="connection_success">Connected Successfully !</string>
     <string name="choose_wifi">Connect to Wi-Fi</string>
+    <string name="wifi_access">Provide Wi-Fi Access</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: Why a capital 'A'?

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (arundhati24) [reply app/src/main/res/values/strings.xml]: Change the string to -> "To search available networks, turn Wi-Fi on"

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -364,13 +364,15 @@
     <string name="digit_five">5</string>
 
     <string name="location_access">Provide Location Access</string>
-    <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby wifi networks available.</string>
+    <string name="location_access_message">Please turn on the device\'s location as it is required to detect the nearby Wi-Fi networks available.</string>
```
Comment (arundhati24) [inline app/src/main/res/values/strings.xml]: unrelated change. Please do not make this change here.

**Comment by codedsun** [comment]
@arundhati24 Check this!

**Comment by codedsun** [comment]
@arundhati24 @batbrain7 please process this PR!

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt [hunk 4]
```diff
@@ -92,6 +97,10 @@ class DeviceConnectPresenter(deviceActivity: DeviceActivity, manager: WifiManage
         checkPermissions = b
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: Create issue to remove single character variable names

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt [hunk 5]
```diff
@@ -92,6 +97,10 @@ class DeviceConnectPresenter(deviceActivity: DeviceActivity, manager: WifiManage
...
     }
 
+    override fun checkWifiEnabled() {
+        isWifiEnabled = mWifiManager.isWifiEnabled
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: Create an issue to remove hungarian notation

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 4 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: https://github.com/fossasia/susi_android/issues/1688 Made

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt (hunk 5 above)
Comment (codedsun) [reply app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectPresenter.kt]: Created https://github.com/fossasia/susi_android/issues/1689

**Comment by codedsun** [comment]
@iamareebjamal Please chck the PR now!
